test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1

```diff
@@ -0,0 +1,400 @@
+        } catch {
+            $_.FullyQualifiedErrorId | Should Be "NoCertificateFound,Microsoft.PowerShell.Commands.UnprotectCmsMessageCommand"
+        }
+    }
+
+    It "Verify failure to extract Ascii armor generates an error" {
+        try {
+            "Hello World" | Unprotect-CmsMessage -ErrorAction Stop
+            throw "No Exception!"
+        } catch {
+            $_.FullyQualifiedErrorId | Should Be "InputContainedNoEncryptedContentIncludeContext,Microsoft.PowerShell.Commands.UnprotectCmsMessageCommand"
+        }
+
+        # Should have round-tripped content
+        $result = "Hello World" | Unprotect-CmsMessage -IncludeContext
+        $result | Should Be "Hello World"
+
+        try {
+            "Hello World" | Get-CmsMessage -ErrorAction Stop
+            throw "No Exception!"
+        } catch {
+            $_.FullyQualifiedErrorId | Should Be "InputContainedNoEncryptedContent,Microsoft.PowerShell.Commands.GetCmsMessageCommand"
+        }
+    }
+
+    It "Verify Unprotect-CmsMessage lets you include context" {
+        $protected = "Hello World" | Protect-CmsMessage -To $certLocation
+        $adjustedProtected = "Pre content`r`n" + $protected + "`r`nPost content"
+
+        $decryptedNoContext = $adjustedProtected | Unprotect-CmsMessage -To $certLocation
+        $decryptedWithContext = $adjustedProtected | Unprotect-CmsMessage -To $certLocation -IncludeContext
+
+        $decryptedNoContext | Should Be "Hello World"
+
+        $decryptedWithContext | Should Match "Pre content"
+        $decryptedWithContext | Should Match "World"
+        $decryptedWithContext | Should Match "Post content"
+    }
+
+    It "Verify Unprotect-CmsMessage treats event logs as a first class citizen" {
+        $protected = "Encrypted Message1","Encrypted Message2" | Protect-CmsMessage -To $certLocation
+        $virtualEventLog = Get-WinEvent Microsoft-Windows-PowerShell/Operational -MaxEvents 1
+        $savedId = $virtualEventLog.Id
+        $virtualEventLog.Message = $protected
+
+        $decrypted = $virtualEventLog | Unprotect-CmsMessage -To $certLocation
+        $decrypted | Should Be "Encrypted Message1`r`nEncrypted Message2"
+
+        $processed = $virtualEventLog | Unprotect-CmsMessage -To $certLocation -IncludeContext
+        $processed.Id | Should Be $savedId
+        $processed.Message | Should Be "Encrypted Message1`r`nEncrypted Message2"
+    }
+
+    It "Verify -DocumentEncryptionCert parameter works" {
+        $foundCerts = Get-ChildItem Cert:\CurrentUser -Recurse -DocumentEncryptionCert
+
+        # Validate they all match the EKU
+        $correctMatching = $foundCerts | ? { $_.EnhancedKeyUsageList[0].ObjectId -eq '1.3.6.1.4.1.311.80.1' }
+        # "All Document Encryption Cert should have had correct EKU"
+        @($foundCerts).Count | Should Be @($correctMatching).Count
+    }
+
+    It "Verify protect message using OutString" {
+        $protected = Get-Process -Id $pid | Protect-CmsMessage -To $certLocation
```

Comment:
I think it's OK. This test is with `feature` tag, and only run in nightly build.